src/components/features/admin/MarkersTable.tsx

```diff
@@ -1,44 +1,63 @@
 import React from 'react'
 import DataTable from '@/components/shared/DataTable'
 import { useTranslation } from 'react-i18next'
-import TableActionsMenu from '@/components/shared/TableActionsMenu'
 import SearchInput from '../SearchInput'
-import { Column } from '@/interfaces/column'
+import { Column } from '@/interfaces/Column'
+import { useSuspenseQuery } from '@tanstack/react-query'
+import { getWarehouseManagementApi } from '@/services/generated-api'
+import { MarkerDto } from '@/services/model'
+import MarkersTableActionsMenu from '../actionsMenu/MarkersTableActionsMenu'
 
 interface Row {
+  id: number
   name: string
-  vendorNumber: number
-  markers: string
   actions: React.ReactNode
 }
 
 export default function MarkersTable() {
   const { t: translate } = useTranslation()
   const [searchTerm, setSearchTerm] = React.useState('')
+  const [page, setPage] = React.useState(0)
+  const [rowsPerPage, setRowsPerPage] = React.useState(10)
 
   const handleSearchChange = (event: React.ChangeEvent<HTMLInputElement>) => {
     setSearchTerm(event.target.value)
   }
 
+  const onPageChange = (newPage: number) => {
+    setPage(newPage)
+  }
+
+  const onRowsPerPageChange = (newRowsPerPage: number) => {
+    setRowsPerPage(newRowsPerPage)
+    setPage(0)
+  }
+
   const columnsData: Column<Row>[] = [
     { key: 'name', title: translate('markers.table.name') },
     { key: 'actions', title: translate('markers.table.actions'), minWidth: 50, align: 'right' }
   ]
 
-  const rowData: Row[] = [
-    {
-      name: 'Motoul',
-      vendorNumber: 1,
-      markers: 'Масло',
-      actions: <TableActionsMenu itemProps={['update', 'delete']} page="markers" />
-    },
-    {
-      name: 'Ferodo',
-      vendorNumber: 2,
-      markers: 'Накладки',
-      actions: <TableActionsMenu itemProps={['update', 'delete']} page="markers" />
+  const { data } = useSuspenseQuery({
+    queryKey: ['markers'],
+    queryFn: () => {
+      return getWarehouseManagementApi().getApiMarkerAll({
+        PageNumber: page + 1,
+        PageSize: rowsPerPage,
+        SearchQuery: searchTerm
+      })
     }
-  ]
+  })
+
+  function transformDataToRows(markers: MarkerDto[]): Row[] {
+    return markers.map((marker: MarkerDto) => ({
+      id: marker.id!,
+      name: marker.name!,
+      actions: <MarkersTableActionsMenu key={marker.id} id={marker.id!} name={marker.name!} />
```

Comment:
just send the whole marker object no need to do each property 1 by 1.

Reply:
Fixed